### PR TITLE
Do not remove directories in purge

### PR DESCRIPTION
**Pull request by liyuqian**
Otherwise, the persistent cache may hold invalid directories and make
subsequent store fail.

The newly added unit tests would fail without the fix.

### CODE REVIEW

File: shell/common/persistent_cache.cc [hunk 1]
```diff
@@ -289,7 +300,7 @@ static void PersistentCacheStore(fml::RefPtr<fml::TaskRunner> worker,
                                   file_name.c_str(),  //
                                   *mapping)           //
         ) {
-          FML_DLOG(WARNING)
+          FML_LOG(ERROR)
```
Comment (liyuqian) [inline shell/common/persistent_cache.cc:303]: When this is reached, we already passed `read_only == false` and `IsValid()` check. So a failure is unexpected and we should print an error. The previous `FML_DLOG` will be optimized out in all builds (including debug build) that are shipped with the Flutter framework.

**Comment by liyuqian** [comment]
Gentle ping...

File: shell/common/persistent_cache.cc [hunk 2]
```diff
@@ -85,15 +85,26 @@ bool PersistentCache::Purge() {
...
-          FML_LOG(INFO) << "Purge persistent cache.";
-          removed.set_value(RemoveFilesInDirectory(*cache_directory));
-        } else {
-          removed.set_value(false);
+  GetWorkerTaskRunner()->PostTask([&removed,
+                                   cache_directory = cache_directory_]() {
+    if (cache_directory->is_valid()) {
+      // Only remove files but not directories.
+      FML_LOG(INFO) << "Purge persistent cache.";
+      fml::FileVisitor recursive_cleanup = [&recursive_cleanup](
```
Comment (jason-simmons) [inline shell/common/persistent_cache.cc:93]: Is this different from calling `VisitFilesRecursively` with a visitor that deletes the file?

Comment (liyuqian) [reply shell/common/persistent_cache.cc]: Yes, `VisitFilesRecursively` with a simple delete visitor would also remove directories. However, I can create a visitor that only deletes file but not directory. Please see my latest patch.

**Review by jason-simmons — APPROVED** [review]

**Pull request merged**